File: oving1.c
#include <pthread.h>
#include <stdio.h>

int i = 0;


void* adder(){
    for(int x = 0; x < 1000000; x++){
        i++;
    }
    return NULL;
}
void* adder2(){
    for(int x = 0; x < 1000000; x++){
        i--;
    }
    return NULL;
}


int main(){
    pthread_t adder_thr;
    pthread_create(&adder_thr, NULL, adder, NULL);
    
    pthread_t adder_thr2;
    pthread_create(&adder_thr2, NULL, adder2,NULL);

    
    pthread_join(adder_thr, NULL);
    pthread_join(adder_thr2,NULL);
    printf("Done: %i\n", i);
    return 0;
    
}

File: oving1.py
from threading import Thread

i=0

def adder1():
  global i
  for x in range(1000000):
    i=i+1
def adder2():
  global i
  for x in range(1000000):
    i=i-1


def main():
  adder_thr= Thread(target = adder1)
  adder_thr.start()
  adder_thr2=Thread(target=adder2)
  adder_thr2.start()
  adder_thr.join()
  adder_thr2.join()
  print("Done: " + str(i))
  
main()
